pkg/accumulator/accumulator.go

```diff
@@ -0,0 +1,221 @@
+package accumulator
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"sniffer/pkg/config"
+	"sniffer/pkg/ebpfeng"
+	evData "sniffer/pkg/ebpfev/v1"
+
+	logger "github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+)
+
+const (
+	CACHE_ACCUMULATOR_SIZE = 10
+)
+
+type containersAccumulator struct {
+	accumulatorDataPerContainer map[string]chan evData.EventData
+	registerMutex               sync.Mutex
+}
+
+type CacheAccumulator struct {
+	accumulatorData                 []map[string][]evData.EventData
+	syncReaderWriterAccumulatorData sync.Mutex
+	firstMapKeysOfAccumulatorData   []string
+	cacheAccumulatorSize            int
+	mainDataChannel                 chan *evData.EventData
+	containersData                  containersAccumulator
+	ebpfEngine                      ebpfeng.EbpfEngineClient
+}
+
+type ContainerAccumulator struct {
+	dataChannel chan evData.EventData
+	containerID string
+}
+
+var myContainerID string
+var cacheAccumulator *CacheAccumulator
+
+func CreateAccumulator() *CacheAccumulator {
```

Comment:
If I understand correctly, the Accumulator is behaving as a Singleton? 
if so, you could simply have a single `GetAccumulator` method, which returns it if `cacheAccumulator != nil`, otherwise creates it.